### PR TITLE
llvm-amdgpu: apply patch from https://github.com/llvm/llvm-project/pull/80071

### PR DESCRIPTION
**Pull request by aweits**
via https://github.com/ROCm/llvm-project/commit/c651b2b0d9d1393fb5191ac3acfe96e5ecc94bbc

issue found in implementation of https://github.com/spack/spack/pull/39666

### CODE REVIEW

File: var/spack/repos/builtin/packages/llvm-amdgpu/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/llvm-amdgpu/package.py]: You could run `spack patch llvm-amdgpu@...` in a binary search to find the minimum version the patch can be applied to.

**Review by adamjstewart — DISMISSED** [review]
Will wait a bit for package maintainers to respond before merging.

File: var/spack/repos/builtin/packages/llvm-amdgpu/package.py
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/llvm-amdgpu/package.py]: Patch applies to all versions for me, but has been fixed in main, so I added an upper bound and removed the lower bound.

**Review by adamjstewart — APPROVED** [review]

**Review by bernhardkaindl — APPROVED** [review]
Checked that the patch is applied with 6.2.1 and it builds.

**Pull request merged**
